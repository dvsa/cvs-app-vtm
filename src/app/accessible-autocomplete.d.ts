declare module 'accessible-autocomplete/dist/accessible-autocomplete.min' {
  interface ElementParam {
    element: HTMLElement;
  }
  interface EnhancedElementParam {
    selectElement: HTMLElement | null;
  }
  interface DefaultParams {
    id?: string;
    source?: Array<any>;
    onConfirm?: (value: string) => void;
    confirmOnBlur?: boolean;
    required?: true;
    defaultValue?: string;
    autoselect?: boolean;
    showAllValues?: boolean;
    dropdownArrow?: Function;
  }

  export interface AutocompleteParams extends DefaultParams, ElementParam {}
  export interface AutocompleteEnhanceParams extends DefaultParams, EnhancedElementParam {}

  export default function accesibleAutocomplete(params: AutocompleteParams): void;
  export function enhanceSelectElement(params: AutocompleteEnhanceParams): void;
}
