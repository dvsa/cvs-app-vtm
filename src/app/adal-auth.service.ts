import { Injectable } from '@angular/core';
import { MsAdalAngular6Service } from 'microsoft-adal-angular6';
// import { MsAdalAngular6Module } from 'microsoft-adal-angular6';


@Injectable({
  providedIn: 'root'
})
export class AdalAuthService {

  constructor(
    private msAdal: MsAdalAngular6Service
  ) { }

  isAuthenticated() {
    return this.msAdal.isAuthenticated
  }

  getProfile() {
    return this.msAdal.userInfo.profile
  }

  getResourceForEndpoint(url) {
    return this.msAdal.GetResourceForEndpoint(url)
  }

  acquireToken(resource) {
    return this.msAdal.acquireToken(resource)
  }

  accessToken() {
    return this.msAdal.accessToken
  }
}
