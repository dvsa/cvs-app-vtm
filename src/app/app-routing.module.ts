import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { MsalGuard } from '@azure/msal-angular';
import { TitleResolver } from './resolvers/title/title.resolver';

const routes: Routes = [
  {
    path: '',
    resolve: { title: TitleResolver },
    children: [
      {
        path: '',
        data: { title: 'Home' },
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/home/home.module').then(m => m.HomeModule)
      },
      {
        path: 'search',
        data: { title: 'Technical record search' },
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/search/search.module').then(m => m.SearchModule)
      },
      {
        path: 'test-records',
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/test-records/test-records.module').then(m => m.TestRecordsModule)
      },
      {
        path: 'tech-records',
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/tech-record/tech-record.module').then(m => m.TechRecordsModule)
      }
    ]
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes, { onSameUrlNavigation: 'reload', scrollPositionRestoration: 'enabled' })],
  exports: [RouterModule]
})
export class AppRoutingModule {}
