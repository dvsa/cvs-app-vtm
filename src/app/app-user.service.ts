import { Injectable } from '@angular/core';

import { UserDetails } from './models/user-details';

@Injectable({ providedIn: 'root' })
export class UserService {
  private user: UserDetails;

  constructor() {}

  setUser(userProfile) {
    this.user = {} as UserDetails;
    this.user.msOid = userProfile.oid;
    this.user.msUser = userProfile.unique_name;
  }

  // TODO: Refactor, remove when auth class is implemented,
  // user retrieved from the store user.details
  getUser() {
    return this.user;
  }

}
