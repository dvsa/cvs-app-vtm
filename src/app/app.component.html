<app-header [username]="userName" (logOutEvent)="logOut()"></app-header>

<div class="govuk-width-container">
  <main class="govuk-main-wrapper" id="main-content" role="main">
    <h1 class="govuk-heading-xl">Default page template</h1>
  </main>
</div>

<app-footer></app-footer>
