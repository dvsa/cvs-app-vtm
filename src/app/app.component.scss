@import 'node_modules/govuk-frontend/govuk/all';

.un-number {
  display: grid;
  grid-template-columns: auto auto;
}

.form-state {
  margin: 20px 20px 20px 10px;
  overflow: auto;
  min-width: 500px;
  max-width: 1248px;
}

.isDisabled {
  color: currentColor;
  cursor: not-allowed;
  opacity: 0.5;
  text-decoration: none;
}

$font-stack: GDSTransportWebsite;
.expander-icon {
  color: #0b0c0c;
  font-size: 15px;
  font-weight: 300;
  letter-spacing: -0.21px;
  line-height: 55px;
  width: 20.8px;
  text-align: center;
}

.ng-animating mat-card mat-expansion-panel-header {
  height: 80px;
}

.ng-animating mat-card div.mat-expansion-panel-content {
  height: 0px;
  visibility: hidden;
}

.no-animate {
  -webkit-transition: none !important;
  transition: none !important;
}

.mt-1 {
  margin-top: 1rem;
}

.mb-1 {
  margin-bottom: 1rem;
}

.no-padding {
  padding: 0 !important;
}

.no-margin {
  margin: 0 !important;
}

dd {
  & ul {
    padding-left: 0%;
  }
}

::ng-deep .mat-expansion-panel-body {
  padding: 0 !important;
}

::ng-deep .auto-pad .mat-expansion-panel-spacing {
  margin: 0 !important;
}

.custom-header .mat-expansion-panel-header-title,
.custom-header .mat-expansion-panel-header-description {
  flex-basis: 0;
}

.custom-header .mat-expansion-panel-header {
  padding: 0;
}

.custom-header,
.open-close-all {
  border-bottom-color: $govuk-border-colour;
  border-bottom-style: solid;
  border-bottom-width: thin;
}

.custom-header .mat-expansion-panel-header-description {
  justify-content: space-between;
  align-items: end;
  flex-direction: row-reverse;
  margin-right: 0%;
}

.disable_ripple:hover {
  background: #fff !important;
  text-decoration-line: underline;
  text-decoration-style: solid;
  text-decoration-color: blue;
}

.rotate_90 {
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -o-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

::ng-deep .custom-header > .mat-expansion-indicator:after {
  color: #777;
  font-weight: bold;
  float: right;
  margin-left: 5px;
}

.expansion-icon {
  color: $govuk-text-colour;
  font-size: 18px;
  font-weight: 300;
  letter-spacing: -0.21px;
  line-height: 55px;
  width: 20.8px;
  text-align: center;
}

.item-technical-record {
  color: $govuk-brand-colour;
  font-size: 36px;
  font-weight: 700;
  letter-spacing: -0.11px;
  line-height: 43px;
  width: 284.96px;
  text-align: left;
  margin-top: 2%;
  white-space: nowrap;
}

.accordion {
  background-color: #eee;
  color: #1a1a1a;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
  transition: 0.4s;
  border-bottom-color: #a5acbd;
  border-top-color: #a5acbd;
  border-top-width: thin;
}

.active,
.accordion:hover {
  background-color: #54fa50;
}

.accordion:after {
  content: '\002B';
  color: #777;
  font-weight: bold;
  float: right;
  margin-left: 5px;
}

.active:after {
  content: '\2212';
  border-bottom-color: #a5acbd;
  border-top-color: #a5acbd;
  border-top-width: thin;
}

.panel {
  padding: 0 18px;
  background-color: white;
  max-height: 0;
  overflow: hidden;
  transition: max-height 0.2s ease-out;
}

.title {
  color: $govuk-text-colour;
  font-size: 48px;
  font-weight: 700;
  letter-spacing: -0.09px;
  line-height: 58px;
  width: 379px;
  text-align: left;
}

.left-pad {
  margin-left: 5%;

  .rectangle {
    background-color: $govuk-border-colour;
    width: 10px;
    height: 60px;
  }

  .grid-container-technical-record-status {
    display: grid;
    grid-column-gap: 10px;
    grid-template-columns: 10px 80px 100px auto;
    width: 50%;
  }

  .grid-item-technical-record-status {
    text-align: left;
    padding-top: 25%;
    font-family: 'GDS Transport', Arial, sans-serif;
  }

  .grid-item-technical-record-status-value {
    text-align: left;
    padding-top: 20%;
    margin-left: -17%;
    font-family: 'GDS Transport', Arial, sans-serif;
  }
}

.auto-pad {
  margin-left: 5%;
  margin-right: 5%;

  &.technical-record-details {
    margin-top: 5%;
  }
}

.records-container {
  width: 100%;
  overflow-x: scroll;
  border: 1px solid rgba(0, 0, 0, 0.03);
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.24), 0 0 2px rgba(0, 0, 0, 0.12);
  margin: 4px;

  & .records-container-title {
    align-content: center;
    align-items: center;
    display: block;
    justify-content: center;
    padding: 8px 20px;
    color: rgba(0, 0, 0, 0.54);
    background: rgba(0, 0, 0, 0.03);

    & .records-container-title-spacer {
      display: flex;
      flex-wrap: wrap;
      color: rgba(0, 0, 0, 0.54);

      > div {
        flex: 0 50%;
      }

      & .record-search-title-text {
        padding-top: 1%;
      }
    }
  }

  & .records-container-body {
    padding: 30px;
    display: flex;

    & ngx-data-table {
      min-width: 130%;

      table {
        width: 100%;
        table-layout: fixed;
      }

      th,
      td {
        overflow: hidden;
        width: 200px;
        text-overflow: ellipsis;
        white-space: nowrap;
      }
    }
  }
}

.mat-expanded {
  border: none !important;
}

.open-close-all {
  display: flex;
  justify-content: flex-end;

  ::ng-deep a {
    color: $govuk-link-colour;
    font-size: 19px;
    font-weight: 500;
    letter-spacing: -0.09px;
    line-height: 45px;
    width: 73px;
    text-align: right;
    flex: 1;
    text-align: right;
    text-decoration: underline;
    cursor: pointer;
  }
}

::ng-deep .item-technical-record:hover,
.mat-expansion-panel-header:hover {
  color: $govuk-link-colour;
  background: none !important;
  text-decoration: underline;
  cursor: pointer;
}

.notes-section {
  font-weight: 100 !important;
}

::ng-deep .govuk-summary-list__key {
  width: 355px !important;
  white-space: nowrap;
}

::ng-deep .main-details {
  padding: 1.5% 0 1.5% 0 !important;
}

@media only screen and (max-width: 600px) {
  .item-technical-record {
    white-space: normal !important;
    font-size: 24px !important;
  }
}

::ng-deep .table-details thead tr th:nth-child(1),
.table-details tbody tr th:nth-child(1) {
  width: 312px !important;
}

::ng-deep .table-details tbody tr > *:not(:first-child) {
  width: 200px !important;
}

.adr-details {
  .grid-container-row {
    display: grid;
    grid-template-columns: auto;
    margin-bottom: 2em;
  }

  .remove-a-un-number {
    float: left;
  }

  .grid-item {
    float: left;
  }

  ::ng-deep h2 {
    margin-top: 2rem !important;
    width: 23rem;
  }

  dd {
    white-space: nowrap !important;

    ::ng-deep &#test-ADR-substancesPermitted.govuk-summary-list__value {
      width: 70% !important;
    }

    @media (max-width: 1313px) {
      &#test-ADR-substancesPermitted {
        height: 3rem !important;

        & span {
          position: absolute;
          white-space: normal !important;
          width: 55vw;
        }
      }
    }
  }

  ::ng-deep .table-tank thead tr th:nth-child(1),
  .table-tank tbody tr td:nth-child(1) {
    width: 23rem !important;
  }

  ::ng-deep .table-tank tbody tr td:nth-child(2) {
    width: 20rem !important;
  }

  ::ng-deep a {
    float: right;
    color: #005ea5;
    font-size: 19px;
    font-weight: 300;
    line-height: 23px;
    width: 102px;
    text-align: right;
    flex: 1;
    text-align: right;
    text-decoration: underline;
    cursor: pointer;
    white-space: nowrap;
  }

  #radioADR {
    display: block;
    margin-top: 1rem;
  }

  #type {
    width: 20rem;
  }

  #permittedDangerousGoods {
    width: 30rem;
  }

  .multiselect {
    height: 13rem;
  }

  .additionalNotes {
    height: 10rem;
  }

  .add-a-guidance-note,
  .add-a-un-number {
    font-weight: 300 !important;
    line-height: 23px !important;
    float: initial !important;
  }

  #adr-more-detail {
    width: 30rem;
  }

  fieldset {
    border: none !important;
  }

  .govuk-summary-list__row:first-child .govuk-summary-list__value:last-child {
    width: 72%;
  }

  display: flow-root !important;
}

a#cancel1,
a#cancel2 {
  color: $govuk-link-colour !important;
  font-family: 'GDS Transport', Arial, sans-serif;
  font-size: 19px;
  font-weight: 400;
  letter-spacing: -0.09px;
  line-height: 40px;
  text-align: right;
  padding-left: 1.5rem;
  cursor: pointer;

  span {
    border-bottom: solid;
    border-bottom-width: 2px;
  }
}

#test-ableToCarry {
  width: 6rem;
}

.govuk-inset-text {
  padding-top: 0 !important;
}

::ng-deep vtm-adr-details-view > div:nth-child(27) {
  display: block !important;
}

.govuk-link,
.govuk-link:visited {
  color: $govuk-link-colour !important;
  font-weight: 500;
  cursor: pointer;
  text-decoration: underline;
}

.govuk-back-link {
  text-decoration: underline;
  cursor: pointer;
}
