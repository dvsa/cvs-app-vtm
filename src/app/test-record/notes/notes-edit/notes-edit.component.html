<div class="govuk-grid-row">
  <div class="govuk-grid-column-full">
    <div class="govuk-form-group" formGroupName="testType">
      <fieldset class="govuk-fieldset">
        <div class="govuk-form-group">
          <div
            class="govuk-grid-row govuk-!-margin-bottom-3"
            [class.govuk-form-group--error]="
              formErrors?.includes('Enter the additionalNotesRecorded')
            "
          >
            <div class="govuk-grid-column-one-half">
              <span id="more-detail-hint" class="govuk-hint">
                Will not be present in the certificate
              </span>
              <span
                *ngIf="formErrors?.includes('Enter the additionalNotesRecorded')"
                class="govuk-error-message"
              >
                <span class="govuk-visually-hidden">Error:</span>
                <span>Enter the carried out during test</span>
              </span>
              <textarea
                class="govuk-textarea"
                id="test-additionalNotesRecorded"
                formControlName="additionalNotesRecorded"
                rows="5"
                [class.govuk-textarea--error]="
                  formErrors?.includes('Enter the additionalNotesRecorded')
                "
              ></textarea>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </div>
</div>
