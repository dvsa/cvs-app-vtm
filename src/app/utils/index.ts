export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


