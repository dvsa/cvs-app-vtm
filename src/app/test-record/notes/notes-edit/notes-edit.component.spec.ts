import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { NotesEditComponent } from './notes-edit.component';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { VIEW_STATE } from '@app/app.enums';
import { FormGroupDirective } from '@angular/forms';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import {TestType} from '@app/models/test.type';

describe('NotesEditComponent', () => {
  let component: NotesEditComponent;
  let fixture: ComponentFixture<NotesEditComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [NotesEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective()
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(NotesEditComponent);
    component = fixture.componentInstance;
    component.currentState = VIEW_STATE.VIEW_ONLY;
    component.testType = {} as TestType;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
