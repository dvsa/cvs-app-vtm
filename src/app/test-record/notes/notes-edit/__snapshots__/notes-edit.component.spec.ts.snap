// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesEditComponent should create 1`] = `
<vtm-notes-edit
  currentState="0"
  testResultChildForm={[Function FormGroupDirective]}
  testType={[Function Object]}
  testTypeGroup="null"
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <div
        class="govuk-form-group"
        formgroupname="testType"
      >
        <fieldset
          class="govuk-fieldset"
        >
          <div
            class="govuk-form-group"
          >
            <div
              class="govuk-grid-row govuk-!-margin-bottom-3"
            >
              <div
                class="govuk-grid-column-one-half"
              >
                <span
                  class="govuk-hint"
                  id="more-detail-hint"
                >
                   Will not be present in the certificate 
                </span>
                
                <textarea
                  class="govuk-textarea"
                  formcontrolname="additionalNotesRecorded"
                  id="test-additionalNotesRecorded"
                  rows="5"
                />
              </div>
            </div>
          </div>
        </fieldset>
      </div>
    </div>
  </div>
</vtm-notes-edit>
`;
